### PR TITLE
Fix invalid JSON for multiple images/containers/pods

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Follow-up on https://github.com/kubernetes-sigs/cri-tools/pull/1486
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/cri-tools/issues/669
#### Does this PR introduce a user-facing change?

```release-note
Output JSON, YAML and go-template slices correctly for `crictl inspect`, `inspectp` and `images`.
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@kubernetes-sigs/cri-tools-maintainers PTAL

**Comment by saschagrunert** [comment]
@kubernetes-sigs/cri-tools-maintainers PTAL

**Comment by kwilczynski** [comment]
/hold

For the one review item.

File: pkg/validate/security_context_linux.go [hunk 1]
```diff
@@ -1579,7 +1579,7 @@ func rootfsPath(info map[string]string) string {
 
 	// The stateDir might have not been created yet. Let's use the parent directory that should
 	// always exist.
-	return filepath.Join(cfg.StateDir, "../")
+	return filepath.Join(cfg.StateDir, "../") //nolint:gocritic
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply pkg/validate/security_context_linux.go:1582]: Yes, I need to drop that change.

**Comment by saschagrunert** [comment]
Rebased to fix the CI.

**Pull request merged**